### ISSUE TITLE
post #356: get rid of `new double`

### ISSUE THREAD

**Issue by drwells** [IBAMR/IBAMR#358]
We should grep through the source and remove any manual allocations of `double` arrays. In all cases
```cpp
auto x = new double[n];
// ...
delete[] x;
```
is just a less convenient version of
```cpp
std::vector<double> x(n);
```

**Comment by fuhuifang** [comment]
That makes sense. I'll make those changes.

**Comment by boyceg** [comment]
Addressed by #393.
